src/squlearn/qrc/base_qrc.py

```diff
@@ -0,0 +1,333 @@
+                params.pop("ml_model_options")
+            initialize_ml_model = True
+
+        if "num_operators" in params or "operator_seed" in params or "operators" in params:
+            if "num_operators" in params:
+                self.num_operators = params["num_operators"]
+                params.pop("num_operators")
+            if "operator_seed" in params:
+                self.operator_seed = params["operator_seed"]
+                params.pop("operator_seed")
+            if "operators" in params:
+                self.operators = params["operators"]
+                params.pop("operators")
+            initialize_observables = True
+            initialize_lowlevel_qnn = True
+            initialize_ml_model = True
+
+        if "param_ini" in params:
+            self.param_ini = params["param_ini"]
+            params.pop("param_ini")
+
+        if "param_op_ini" in params:
+            self.param_op_ini = params["param_op_ini"]
+            params.pop("param_op_ini")
+
+        if initialize_observables:
+            self._initialize_observables()
+            initialize_lowlevel_qnn = True
+            initialize_ml_model = True
+
+        if initialize_lowlevel_qnn:
+            self._initialize_lowlevel_qnn()
+            # Reinitialize parameters if the number of parameters has changed
+            initialize_parameters = (
+                self.param_ini is None or len(self.param_ini) != self._qnn.num_parameters
+            )
+            initialize_parameters_obs = (
+                self.param_op_ini is None
+                or len(self.param_op_ini) != self._qnn.num_parameters_observable
+            )
+
+            self._initialize_parameters(initialize_parameters, initialize_parameters_obs)
+
+        if initialize_ml_model:
+            self._initialize_ml_model()
+
+        if "caching" in params:
+            self.caching = params["caching"]
+            params.pop("caching")
+            self._qnn.result_caching = self.caching
+
+        # Set encoding_circuit parameters
+        ec_params = params.keys() & self.encoding_circuit.get_params(deep=True).keys()
+        if ec_params:
+            self.encoding_circuit.set_params(**{key: params[key] for key in ec_params})
+            if self.encoding_circuit.num_parameters != len(self.param_ini):
+                self._initialize_parameters()
+
+        # Set qnn parameters
+        qnn_params = params.keys() & self._qnn.get_params(deep=True).keys()
+        if qnn_params:
+            self._qnn.set_params(**{key: params[key] for key in qnn_params})
+            if self._qnn.num_parameters != len(
+                self.param_ini
+            ) or self._qnn.num_parameters_observable != len(self.param_op_ini):
+                self._initialize_parameters()
+
+        if "parameter_seed" in params:
+            self.parameter_seed = params["parameter_seed"]
+            params.pop("parameter_seed")
+            self._initialize_parameters()
+
+        return self
```

Comment:
This is unnecessarily complicated. Can you please adapt to use bitwise and `&` and more grouped `if` statements. If you want to, i can help with this.